### PR TITLE
XML Entity Encoding Fix, Redirect Fix, Documentation Update

### PR DESCRIPTION
**Pull request by daniel-dgi**

### CODE REVIEW

**Comment by willtp87** [comment]
Talked to Danny about temp files and some warning about the entity replacement.  Going to use managed files and the warning can go in the README.

File: README.md [hunk 1]
```diff
@@ -2,21 +2,75 @@
...
+
+Basic Algorithm:
+* [Install Zorba](http://www.zorba.io/documentation/latest/zorba/install)
+* [Compile libxdiff from source (./configure, make, make install)](http://www.xmailserver.org/xdiff-lib.html)
+* [Install PHP xdiff extension](http://www.php.net/manual/en/xdiff.setup.php)
+* Install GeSHi using the Libraries API (put geshi folder in sites/all/libraries)
+* [Install Islandora Object Lock Module](https://github.com/discoverygarden/islandora_object_lock)
+* Install Islandora Xquery Module
+
+```bash
```
Comment (willtp87) [inline README.md]: Talked to Danny, moving this out of the README and just going to link it.

File: includes/cancel.inc [hunk 2]
```diff
@@ -29,5 +29,6 @@ function islandora_xquery_cancel_results($batch_id) {
     ->execute();
 
   // Redirect.
-  drupal_goto($redirect);
+  $url = drupal_parse_url($redirect);
```
Comment (willtp87) [inline includes/cancel.inc:32]: I think this will pass through an unused key of path to drupal_goto.

File: includes/apply.inc [hunk 3]
```diff
@@ -87,25 +88,24 @@ function islandora_xquery_apply_query_batch_operation($batch_id, array &$context
...
-        $ignored_objects[] = $result->pid;
-        $include = FALSE;
-      }
-      elseif ($created_date > $batch_time) {
-        $ignored_objects[] = $result->pid;
-        $include = FALSE;
-      }
-      islandora_object_lock_set_object_lock($result->pid);
-      islandora_xquery_apply_query_batch_apply_diff($result, $context);
+
```
Comment (willtp87) [inline includes/apply.inc:91]: Removing the module_exists from around this code makes it have a hard dependency that is not noted in the .info.  Either the module_exists call should persist to maintain the dependency as soft or the .info should be updated.

File: includes/cancel.inc (hunk 2 above)
Comment (daniel-dgi) [inline includes/cancel.inc:32]: What's wrong with this?  More or less straight jacked this from drupal_parse_url() documentation.  Using it the same way with drupal_goto as it is used with url() in the example.  https://api.drupal.org/api/drupal/includes!common.inc/function/drupal_parse_url/7

File: includes/apply.inc (hunk 3 above)
Comment (daniel-dgi) [inline includes/apply.inc:91]: Was only mentioned in the README.  I'm making it a hard dependency in the .info file.

File: includes/cancel.inc (hunk 2 above)
Comment (willtp87) [inline includes/cancel.inc:32]: Talked with Danny about this.  It's weird to pass extra unused arguments to a function, but if Drupal has it in their examples then it is unlikely to be sending warnings around any time soon.

**Pull request merged**